src/plugins/intel_cpu/src/nodes/kernels/scaled_attn/xattention.hpp

```diff
@@ -150,7 +151,8 @@ struct Xattn {
             return;
         }
 
-        if (_xattn_gemm.empty()) {
+        if (_xattn_gemm.empty() || _k_len != B) {
```

Comment:
The kernels are not changed with `batch in tokens`, the changed should be those intermediate buffers, I think you could separate the logic between kernel creation and buffer re-allocation.

Reply:
Since ldc changes when sequence length changes, the kernel need to be recreated. I think we can keep it as is and figure out how to optimize it in multi-batch case.